test/integration/testpaths/testpath_volume_cuncurrent_snapshots.py

```diff
@@ -0,0 +1,827 @@
+
+            thread_pool.append(t2)
+
+            for thread in thread_pool:
+                thread.start()
+
+            for thread in thread_pool:
+                thread.join()
+
+        except Exception as e:
+            raise Exception(
+                "Warning: Exception unable to start thread : %s" %
+                e)
+
+        snapshots = list_snapshots(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+            listall=True
+        )
+
+        for snaps in self.snapshot_pool:
+            self.assertTrue(snaps.id in any(
+                snp.id) for snp in snapshots)
+
+        for snapshot in self.snapshot_pool:
+            self.assertTrue(
+                is_snapshot_on_nfs(
+                    self.apiclient,
+                    self.dbclient,
+                    self.config,
+                    self.zone.id,
+                    snapshot.id))
+
+        for snap in self.snapshot_pool:
+            snap.delete(self.apiclient)
+
+        self.snapshot_pool = []
+
+        # Step 7
+        thread_pool = []
+        try:
+            for i in range(2):
+                destinationHost = Host.listForMigration(
+                    self.apiclient,
+                    virtualmachineid=self.vm_pool[i].id)
+                t = Thread(target=MigrateRootVolume,
+                           args=(self,
+                                 self.vm_pool[i],
+                                 destinationHost[0]))
+
+                thread_pool.append(t)
+
+            for t in thread_pool:
+                t.start()
+
+            for t in thread_pool:
+                t.join()
+
+            for vm in self.vm_pool:
+                if vm.state != "Stopped":
+                    vm.stop(self.apiclient)
+
+            t_pool = []
+            for vm in self.vm_pool[:2]:
+                t = Thread(
+                    target=CreateSnapshot,
+                    args=(
+                        self,
+                        self.root_pool[0],
+                        False))
+                t_pool.append(t)
+
+            for t in t_pool:
+                t.start()
+
+            for t in t_pool:
+                t.join()
+
+        except:
+            self.debug("Error: unable to start thread")
+
+        snapshots = list_snapshots(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+            listall=True
+        )
+
+        for snaps in self.snapshot_pool:
+            self.assertTrue(snaps.id in any(
+                snp.id) for snp in snapshots)
+
+        for snapshot in self.snapshot_pool:
+            self.assertTrue(
+                is_snapshot_on_nfs(
+                    self.apiclient,
+                    self.dbclient,
+                    self.config,
+                    self.zone.id,
+                    snapshot.id))
+
+        for snap in self.snapshot_pool:
```

Comment:
use "snapshot" instead of snap. Make changes at all appropriate places so that variable names will convey more information about the value it is holding.